Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -298,130 +311,58 @@ class NexusGeometrySaveTest : public CxxTest::TestSuite {
+            true, false, true);
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*instr.first);
 
-  void
-  test_bank_absolute_rotation_is_equal_to_detector_in_bank_absolute_rotation() {
-
-    auto instrument =
-        ComponentCreationHelper::createTestInstrumentRectangular2(1, 10);
-
-    auto beamline =
-        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
-    auto &compInfo = (*beamline.first);
-
-    // set abosolute bank rotation (rotate about y)
-    Eigen::Quaterniond absoluteBankRotation(
-        Eigen::AngleAxisd(M_PI / 4, Eigen::Vector3d(0, 1, 0)));
-
-    Eigen::Quaterniond zeroRotation(
-        Eigen::AngleAxisd(0, Eigen::Vector3d(0, 0, 0)));
-
-    // relative position of detector
-    auto relativeDetectorRotation =
-        Mantid::Kernel::toQuaterniond(compInfo.relativeRotation(0));
+    ScopedFileHandle fileResource("check_no_sample_throws_test_file.hdf5");
+    auto destinationFile = fileResource.fullPath();
 
-    TS_ASSERT(relativeDetectorRotation.isApprox(zeroRotation));
+    TS_ASSERT(compInfo.hasDetectorInfo()); // rule out throw by no detector
+                                           // info.
+    TS_ASSERT(compInfo.hasSource());       // rule out throw by no source.
+    TS_ASSERT_THROWS(saveInstrument(compInfo, destinationFile),
+                     std::invalid_argument &);
   }
 
-  void test_detector_position_is_expected_value_after_rotation() {
+  void test_instrument_without_source_throws() {
 
-    // 45 degree rotation around z.
-    const Quat bankRotation(45, V3D(0, 0, 1));
-    const Quat detRotation(45, V3D(0, 0, 1));
+    auto const &instrument =
+        ComponentCreationHelper::createInstrumentWithOptionalComponents(
+            false, true, true);
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*instr.first);
 
-    auto instrument =
-        ComponentCreationHelper::createSimpleInstrumentWithRotation(
-            Mantid::Kernel::V3D(0, 0, -10), Mantid::Kernel::V3D(0, 0, 0),
-            Mantid::Kernel::V3D(0, 0, 10), bankRotation, detRotation);
-
-    auto beamline =
-        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
-    const auto &compInfo = *std::get<0>(beamline);
-
-    const auto actualBankRotation =
-        compInfo.rotation(compInfo.indexOfAny("detector-stage"));
-    const auto actualDetRotation = compInfo.rotation(0);
+    ScopedFileHandle fileResource("check_no_source_throws_test_file.hdf5");
+    auto destinationFile = fileResource.fullPath();
 
-    TS_ASSERT_EQUALS(actualBankRotation, bankRotation);
-    TS_ASSERT_EQUALS(actualDetRotation, detRotation * bankRotation);
+    TS_ASSERT(compInfo.hasDetectorInfo()); // rule out throw by no detector
+                                           // info.
+    TS_ASSERT(compInfo.hasSample());       // rule out throw by no sample.
```

Comment:
By extension, you should probably check that `TS_ASSERT(!compInfo.hasSource())` to explain why we therefore expect `saveInstrument` to throw.